spec/std/http/cookie_spec.cr

```diff
@@ -45,6 +45,29 @@ module HTTP
           HTTP::Cookie.new("x", %(foo\rbar))
         end
       end
+
+      describe "with a security prefix" do
+        it "raises on invalid cookie with prefix" do
+          expect_raises ArgumentError, "Invalid cookie name. Has '__Secure-' prefix, but is not secure." do
+            HTTP::Cookie.new("__Secure-foo", "bar", secure: false)
+          end
+
+          expect_raises ArgumentError, "Invalid cookie name. Does not meet '__Host-' prefix requirements." do
+            HTTP::Cookie.new("__Host-foo", "bar", domain: "foo")
```

Comment:
It would be helpful to name the prefix requirements explicitly: secure, no domain, path `/` (don't necessarily need to point out which one exactly failed, but that would be a nice enhancement).